src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs

```diff
@@ -23,7 +27,25 @@ private void MigrateCode()
 
         public void ChangeCode()
         {
+            SavePythonMigrationBackup();
             PythonNode.MigrateCode(NewCode);
         }
+
+        private void SavePythonMigrationBackup()
+        {
+            var path = GetPythonMigrationBackupPath();
+            if (File.Exists(path))
+                return;
+
+            DynamoViewModel.SaveAs(path, true);
```

Comment:
At this point, would it make sense to somehow notify the user that a backup was created? Otherwise they might never realize of their existence. What do you think @Amoursol ?

Reply:
Definitely notify users, and we should make sure it persists between versions also.

Reply:
How would we like to notify users ? 

1. simple message box 
2. custom notification (like those on Dynamo startup)
3. Dynamo notification system (does anyone see or look at those ?)

also @Amoursol , what do you mean by `persist between versions` ?

Reply:
@radumg Can we look to add a `simple message box` that states to the user a message stating that it has been created and it's location (On the proviso they want to delete at some point). 

In terms of persist between versions: If a user uses Dynamo 2.7 Sandbox, then downloads 2.8 Sandbox then we should honour their choices and note this in the message above.

Reply:
Are there any choices to honor here? Also moving preferences between 2.7 and 2.8 etc - is a whole other can of worms.

Reply:
if we can avoid that 🥫🐍 then we should 😁 
Don't think there's anything to persist between versions anyway, backup location is version-agnostic. Unless you mean disabling this backup notification if they've also disabled the other migration warning?

this is what it looks like now :
![image](https://user-images.githubusercontent.com/11439624/88813492-49785800-d1b9-11ea-8f50-e7683133ed49.png)

The text is as below, with `{0}` replaced with path.
> Before applying the Python code migration, a backup of this graph has been saved to the configured backup location : {0}